Sources/Basic/DictionaryExtensions.swift

```diff
@@ -0,0 +1,20 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+extension Dictionary {
```

Comment:
Let's put this in a separate PR, I think we all should review extensions to core types in Basic (partly so that we are all aware of which ones we add, and can use them consistently where appropriate).